pkg/controller/scheduledjob/controller.go

```diff
@@ -58,11 +59,11 @@ type ScheduledJobController struct {
 	recorder   record.EventRecorder
 }
 
-func NewScheduledJobController(kubeClient *client.Client) *ScheduledJobController {
+func NewScheduledJobController(kubeClient *client.Client, eventClient coreclient.EventsGetter) *ScheduledJobController {
 	eventBroadcaster := record.NewBroadcaster()
 	eventBroadcaster.StartLogging(glog.Infof)
 	// TODO: remove the wrapper when every clients have moved to use the clientset.
-	eventBroadcaster.StartRecordingToSink(kubeClient.Events(""))
+	eventBroadcaster.StartRecordingToSink(eventClient.Events(""))
```

Comment:
Is it intentional that you add a separate event client in some controller, but not all?